examples/mnist/convert_mnist_data.cpp

```diff
@@ -6,12 +6,22 @@
 // The MNIST dataset could be downloaded at
 //    http://yann.lecun.com/exdb/mnist/
 
+#ifdef USE_GFLAGS
 #include <gflags/gflags.h>
-#include <glog/logging.h>
+#endif   // USE_GFLAGS
+#ifdef USE_GLOG
+#include "glog/logging.h"
+#else
+#include "caffe/util/glog_alternate.hpp"
+#endif  // USE_GLOG
```

Comment:
It's really boring to repeat the guarding macro checks. In common.hpp or somewhere else, define such things
```cpp
#ifdef USE_STH
#define STH_HEADERS <sth/sth.h>
#else
#define STH_HEADERS <sth/else/sth_else.hpp>
#endif
```